py-polars/pyproject.toml

```
@@ -40,10 +40,12 @@ Changelog = "https://github.com/pola-rs/polars/releases"
 [project.optional-dependencies]
 # NOTE: keep this list in sync with show_versions() and requirements-dev.txt
 adbc = ["adbc_driver_manager", "adbc_driver_sqlite"]
-async = ["nest_asyncio"]
+async = ["gevent", "nest_asyncio"]
 cloudpickle = ["cloudpickle"]
 connectorx = ["connectorx >= 0.3.2"]
+database = ["adbc_driver_manager", "adbc_driver_sqlite", "connectorx >= 0.3.2", "nest_asyncio", "pandas", "sqlalchemy"]
 deltalake = ["deltalake >= 0.15.0"]
+excel = ["fastexcel >= 0.9", "openpyxl >= 3.0.0", "xlsx2csv >= 0.8.0", "xlsxwriter"]
```

For the 'groups', I think we can re-use the requirements we already defined (same as we currently do for `polars[all]`. That way we only have to define version constraints in a single place:

```suggestion
excel = ["polars[fastexcel,openpyxl,xlsx2csv,xlsxwriter"]
```